connectors/sources/github.py

```diff
@@ -1185,28 +1439,86 @@ async def get_docs(self, filtering=None):
         Yields:
             dict: Documents from GitHub.
         """
-        async for repo in self._fetch_repos():
-            yield repo, None
-            repo_name = repo.get("nameWithOwner")
-            default_branch = (
-                repo.get("defaultBranchRef", {}).get("name")
-                if repo.get("defaultBranchRef")
-                else None
-            )
+        if filtering and filtering.has_advanced_rules():
+            advanced_rules = filtering.get_advanced_rules()
+            for rule in advanced_rules:
+                repo = await anext(
+                    self._get_configured_repos(configured_repos=[rule["repository"]])
+                )
+                yield repo, None
+                repo_name = repo.get("nameWithOwner")
+
+                if pull_request_query := rule["filter"].get("pr"):
+                    query_status, pull_request_query = self._filter_rule_query(
+                        repo=repo_name,
+                        query=pull_request_query,
+                        query_type="pr",
+                    )
+                    if query_status:
+                        async for pull_request in self._fetch_pull_requests(
+                            repo_name=repo_name,
+                            response_key=("search",),
+                            filter_query=pull_request_query,
+                            query=GithubQuery.SEARCH_QUERY.value,
+                        ):
+                            yield pull_request, None
+                    else:
+                        self._logger.warning(
+                            f"Skipping pr for rule: {pull_request_query}"
```

Comment:
`query_status` only returns boolean value whenever the query is not related to their object then it will return False otherwise True. Example: When issue query is passed in pr key then `query_status` will be False. IMO `query_status` shouldn't be part of logger as it will also False in skipping logger.